analyses/rna-seq-protocol-dge/03-protocol-dge-seg.R

```diff
@@ -0,0 +1,315 @@
+                   'differential gene expression ', plot_nm_str,
+                   ' normalized\n',
+                   'exactTest p-values\n',
+                   sum(et_top_tags$table$PValue < 0.05),
+                   ' genes have p-value < 0.05\n',
+                   sum(et_top_tags$table$PValue >= 0.05),
+                   ' genes have p-value >= 0.05\n',
+                   sum(et_top_tags$table$FDR < 0.05),
+                   ' genes have BH FDR < 0.05\n',
+                   sum(et_top_tags$table$FDR >= 0.05),
+                   ' genes have BH FDR >= 0.05')) +
+    theme(text = element_text(size=15))
+
+ggsave(file.path(plot_outdir,
+                 'stranded_vs_polya_dge_exact_test_pvals_histogram.png'),
+       dpi = 300, plot = p, width = 8, height = 7)
+#------------ Run QL F test ------------------------------------
+print(paste0('Run differential gene expression QL F-test on',
+             ' poly-A vs stranded RNA-seq...'))
+ql_fit <- glmQLFit(counts_object, design)
+png(file.path(plot_outdir, 'glm_ql_fit_deviances.png'))
+plotQLDisp(ql_fit)
+n_null_dev <- dev.off()
+qlft <- glmQLFTest(ql_fit)
+# get p-value table
+qlft_top_tags <- topTags(qlft, adjust.method = "BH", n = Inf)
+qlft_out_df <- qlft_top_tags$table
+stopifnot(identical(colnames(qlft_out_df),
+                    c("logFC", "logCPM", "F", "PValue", "FDR")))
+colnames(qlft_out_df) <- c("stranded_over_polya_logFC",
+                           "average_logCPM", "F", "PValue", "BH FDR")
+write.csv(qlft_out_df,
+          file=file.path(table_outdir,
+                         'stranded_vs_polya_dge_ql_ftest_res.csv'))
+# plot and save p-value histogram
+p <- ggplot(qlft_top_tags$table, aes(x=PValue)) +
+    geom_histogram(binwidth = 0.05, center = 0.025) +
+    theme_classic() +
+    scale_x_continuous(expand = expand_scale(mult = c(0, 0.02))) +
+    scale_y_continuous(expand = expand_scale(mult = c(0, 0.05))) +
+    xlab(paste0('stranded vs poly-A RNA-seq DGE ', plot_nm_str,
+                ' QL F-test p-value')) +
+    ylab('Gene count') +
+    ggtitle(paste0('Histogram of stranded vs poly-A RNA-seq\n',
+                   'differential gene expression ', plot_nm_str,
+                   ' normalized\n',
+                   'QL F-test p-values\n',
+                   sum(qlft_top_tags$table$PValue < 0.05),
+                   ' genes have p-value < 0.05\n',
+                   sum(qlft_top_tags$table$PValue >= 0.05),
+                   ' genes have p-value >= 0.05\n',
+                   sum(qlft_top_tags$table$FDR < 0.05),
+                   ' genes have BH FDR < 0.05\n',
+                   sum(qlft_top_tags$table$FDR >= 0.05),
+                   ' genes have BH FDR >= 0.05')) +
+    theme(text = element_text(size=15))
+
+ggsave(file.path(plot_outdir,
+                 'stranded_vs_polya_dge_ql_ftest_pvals_histogram.png'),
+       dpi = 300, plot = p, width = 8, height = 7)
+#------------ Run DESeq2 nbinomWaldTest ------------------------
```

Comment:
Is this section from lines 255-315 distinct from `04-deseq2-protocol-dge.R`? Looks as though the only variation is the non-default estimation of dispersions.

Reply:
Sorry for the confusion. This section is distinct from `04-deseq2-protocol-dge.R`.

In addition to the differences on dispersion estimation methods, the normalization methods are also different. `03-protocol-dge-seg.R` uses TMM normalized CPM or UQ-pgQ2 normalized read counts for DESeq2 DGE analysis, whereas `04-deseq2-protocol-dge.R` uses the DESeq2 default normalization Relative Log Expression (RLE) method.